packages/cx/src/widgets/grid/Grid.js

```diff
@@ -295,7 +296,8 @@ export class Grid extends Container {
          data.sorters = [sorter];
       }
 
-      if (!isNonEmptyArray(data.sorters) && this.defaultSortField) {
+      let skipSorting = this.clearableSort && this.allowClearSort;
```

Comment:
There is one minor nitpick. The state of `allowClearSort` should be stored on the `instance` instead of the `widget`, as the widget can be instantiated multiple times (within a Repeater). 

While there, it would be better to call it `skipDefaultSorting` instead of `skipSorting`...

Also, it might be better to rename `allowClearSort` to `disableDefaultSort`.